data/core/init.lua

```diff
@@ -535,16 +535,18 @@ end
 
 function core.load_plugins()
   local no_errors = true
-  for _, root_dir in ipairs {DATADIR, USERDIR} do
+  local loaded = {}
+  for _, root_dir in ipairs {USERDIR, DATADIR} do
     local files = system.list_dir(root_dir .. "/plugins")
     for _, filename in ipairs(files or {}) do
       local basename = filename:gsub(".lua$", "")
```

Comment:
Please replace `filename:sub(...)` with `filename:match("(.+)%.lua$")` and check below basename is not nil to exclude files whose extension is not "lua".

Reply:
Wouldn't it be better if we used `common.basename` here instead? Also, it would be nice if we could support plugins structured like modules `plugins/myplugin/init.lua`. `require` already provides the functionality to load such modules, we'd only need to add another condition to the loader.

Reply:
Nice idea, we may check for:

- files whose name ends in ".lua"
- directories (supposing the may contain an init.lua file). May be proactively check if init.lua is there to avoid trying to "require" directories with random stuff but not really required.